lib/jekyll-webp/webpExec.rb

```diff
@@ -27,13 +27,25 @@ def self.run(quality, input_file, output_file)
         full_path = File.join(gem_root, bin_path, exe_name)
 
         # Construct the full program call
-        cmd = "\"#{full_path}\" -quiet -mt -q #{quality.to_s} \"#{input_file}\" -o \"#{output_file}\""
+        cmd = "\"#{full_path}\" -quiet -mt #{flags} \"#{input_file}\" -o \"#{output_file}\""
         
         # Execute the command
-        stdin, stdout, stderr = Open3.popen3(cmd)
+        exit_code = 0
+        error = ""
+        output = ""
+        Open3.popen3(cmd) do |stdin, stdout, stderr, wait_thr|
+          stdin.close # we don't pass any input to the process
+          output = stdout.gets
+          error = stderr.gets
+          exit_code = wait_thr.value
+        end
+
+        if exit_code != 0
+          Jekyll.logger.error("WebP:","cwebp returned #{exit_code} with error #{error}")
```

Comment:
This is good, however we need to cater to the average user. A non-technical error message would be better in this case. E.g. "WebP" "Conversion for image XXXX failed, no webp version could be created for this image" then the actual error message can be printed to debug for the devs.